src/main/java/seedu/todo/model/TodoModel.java

```diff
@@ -151,26 +146,34 @@ public ImmutableTask delete(int index) throws ValidationException {
         uniqueTagCollection.notifyTaskDeleted(taskToDelete);
         return todoList.delete(taskIndex);
     }
+    
+    @Override
+    public List<ImmutableTask> deleteAll() throws ValidationException{
+        saveUndoState();
+        List<Integer> indexes = new ArrayList<>();
+        for (int i = 1 ; i <= getObservableList().size() ; i++) {
+            indexes.add(getTaskIndex(i));
+            ImmutableTask taskToDelete = tasks.get(getTaskIndex(i));
+            uniqueTagCollection.notifyTaskDeleted(taskToDelete);
```

Comment:
What happens if a validation error occurs? Not all tasks are deleted now... 😆 
Only inform tags after all tasks are deleted.
Do it what you previously did (second for loop).